src/main/java/com/theelixrlabs/healthcare/service/PatientService.java

```diff
@@ -114,4 +133,25 @@ public PatientDTO getPatientById(UUID patientId) throws CustomException {
                 .patientAadhaarNumber(patientModel.getPatientAadhaarNumber())
                 .build();
     }
+
+    /**
+     * Deletes a patient by their ID, if conditions are met.
+     *
+     * @param patientId The ID of the patient to delete.
+     * @return A success message upon successful deletion.
+     * @throws CustomException If the patient is not found or is currently assigned to a doctor.
+     */
+    public String deletePatientById(String patientId) throws CustomException {
+        UUID validPatientId = validateUUID(patientId);
+        Optional<PatientModel> patientModelOptional = patientRepository.findById(validPatientId);
+        if (patientModelOptional.isEmpty())
+            throw new CustomException(PatientConstants.PATIENT_NOT_FOUND_KEY, messageSource);
+        List<DoctorPatientAssignmentModel> patientAssignmentList = doctorPatientAssignmentRepository.findByPatientId(validPatientId);
+        if (!patientAssignmentList.isEmpty()) {
+            throw new CustomException(PatientConstants.PATIENT_DELETION_FAILED_ASSIGNED_TO_DOCTOR, messageSource);
+        }
+        patientRepository.deleteById(validPatientId);
+        Object[] dynamicArguments = {patientId};
+        return messageSource.getMessage(PatientConstants.PATIENT_DELETE_SUCCESS_MESSAGE, dynamicArguments, Locale.getDefault());
```

Comment:
lets make use of new Object[]{patientId} directly in place of dynamicArguments as you are passing only patientId

Reply:
Resolved.